pkg/elasticsearch/elasticsearch_service.go

```diff
@@ -23,3 +30,41 @@ func (es ElasticSearchService) GetStatus() int {
 
 	return service.ServiceStatusPending
 }
+
+func (es ElasticSearchService) SetCertificateToService(certDto *dtos.CertificateDto) (interface{}, error) {
+	elasticInstance := es.crdInstance
+	tlsCert := map[string][]byte{
+		"ca.crt":  []byte(certDto.CaCrt),
```

Comment:
Since the DTO is transfered as json the encoding should rely on base64.  []byte() should not work with the frontend. Did you try it out?

Reply:
certDto is defined as {CaCrt: string, TlsCrt: string, TlsKey: string}. As u have said before, user would give certificate info as string, right ? I have already implemented the test for setting certificate for Elasticsearch service. In test/integration_tests/provider/elasticsearch_provider_test.go from lien 140-164, i have write the test to check whether  the service status stays ok after setting the certificate

Reply:
Yes it comes from the user and yes it comes as a string. This is the common way to transfer binary data in json, but it is encoded as base64 to overcome charset problems.  []byte(...) does not decode base64.

Reply:
So, the input for fields would be not in form like --BEGIN CERTIFICATE---............---END CERTIFICATE------- but in form of Base64 ?